app/services/users/login_service.rb

```diff
@@ -61,18 +63,25 @@ def set_autologin_cookie
       return unless Setting::Autologin.enabled?
 
       # generate a key and set cookie if autologin
-      expires_on =  Setting.autologin.days.from_now.beginning_of_day
-      token = Token::AutoLogin.create(user:, data: session_identification, expires_on:)
+      token = Token::AutoLogin.create(user:, data: token_session_information)
       cookie_options = {
         value: token.plain_value,
         # The autologin expiry is checked on validating the token
         # but still expire the cookie to avoid unnecessary retries
-        expires: expires_on,
+        expires: token.expires_on,
         path: OpenProject::Configuration["autologin_cookie_path"],
         secure: OpenProject::Configuration.https?,
         httponly: true
       }
       cookies[OpenProject::Configuration["autologin_cookie_name"]] = cookie_options
+
+      # Tie the current session to this autologin token already
+      assign_autologin_session(token.id)
+    end
+
+    def assign_autologin_session(token_id)
+      session_id = ::Sessions::UserSession.where(session_id: session.id.private_id).pick(:id)
```

Comment:
:question: Would it at this point be possible for the `session_id` to be nil because we didn't find a corresponding user session? If so, the creation below would fail due to a violation of the `NOT NULL` constraint...